versioned/tiered/tiered-impl/src/main/java/org/projectnessie/versioned/impl/TieredVersionStoreConfig.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.impl;
+
+import org.immutables.value.Value.Default;
+import org.immutables.value.Value.Immutable;
+import org.projectnessie.versioned.impl.ImmutableTieredVersionStoreConfig.Builder;
+
+@Immutable
+public interface TieredVersionStoreConfig {
```

Comment:
I generally don't think these should be end user configurable. I suggest we come up with things that should be configured versus those that generally shouldn't be and namespace them separately. Additionally, this exposes some internal detail. For example, I think we should actually adjust so we only collapse on save on every N commits or time, which makes the first option here a bit misleading.

Reply:
Yea, concurrent commits and especially the kicked-off collapses are tricky. The collapses can be synchronized (i.e. only one collapse per branch), which already works fine. I implemented a mechanism to ensure that only one collapse per Nessie instance is running for each branch for perf-testing (the `Serialize collapse-branch-commits per branch` commit in my `trace-commit` branch + a follow-up that eliminates some more read operations in the commit-code-path). The retry-mechanism(s) used by commit + collapse should have reasonable defaults, but IMO should be tunable, maybe with a configurable retry/back-off strategy - but it's a complicated thing to tune and requires a lot of in-depth knowledge. I'd like to have those at least for now, but maybe prefix those with `internal` or so. WDYT?

Reply:
> but it's a complicated thing to tune and requires a lot of in-depth knowledge

This to me means they shouldn't be exposed as parameters almost by definition. if I had a dollar for every time I have seen exposing internal details as config result in poor performance. bugs, correctness errors and generally pain...

an `internal` prefix won't discourage or even slow most people down. The best speed bump i can think of is to make people recompile. Which given the likelihood of people needing to change these parameters and the type of people that would change them isn't to me such a big deal.